metadata/src/lib.rs

```diff
@@ -132,6 +138,11 @@ impl Metadata {
         })
     }
 
+    /// Returns custom user defined types
+    pub fn custom_metadata(&self) -> &CustomMetadata<PortableForm> {
+        &self.custom
+    }
```

Comment:
I think we should hide the internals (frame-metadata stuff) a little better here and have an api more like:

```rust
let custom = metadata.custom();

let Some(value) = custom.get("Foo") else { return };

let type_id: u32 = value.ty();
let data: &[u8] = value.bytes();

// so maybe thats implemented something like:

pub struct CustomMetadata {
    // We can leave the frame_metadata map as is:
    values: BTreeMap<String, frame_metadata::CustomValueMetadata<T>>
}

pub struct CustomMetadataValue<'a> {
    type_id: u32,
    data: &'a Vec<u8>
}

impl CustomMetadata {
    pub fn get(&'a self, &str) -> Option<CustomMetadataValue<'a>>
}

impl CustomMetadataValue {
    pub fn ty(&self) -> u32;
    pub fn bytes(&Self) -> &[u8]
}
```